ctapipe/reco/sklearn.py

```diff
@@ -0,0 +1,590 @@
+
+        return prediction, valid
+
+    def _table_to_y(self, table, mask=None):
+        y = super()._table_to_y(table, mask=mask)
+
+        if self.log_target:
+            if np.any(y <= 0):
+                raise ValueError("y contains negative values, cannot apply log")
+
+            return np.log(y)
+        return y
+
+
+class SKLearnClassficationReconstructor(SKLearnReconstructor):
+    """
+    Base class for classification tasks
+    """
+
+    model_cls = Enum(
+        SUPPORTED_CLASSIFIERS.keys(),
+        default_value=None,
+        allow_none=True,
+        help="Which scikit-learn regression model to use.",
+    ).tag(config=True)
+
+    invalid_class = Integer(
+        default_value=-1, help="The label to fill in case no prediction could be made"
+    ).tag(config=True)
+
+    positive_class = Integer(
+        default_value=1,
+        help=(
+            "The label value of the positive class."
+            " ``prediction`` values close to 1.0 mean the event"
+            " belonged likely to this class."
+        ),
+    ).tag(config=True)
+
+    def _predict(self, key, table):
+        if key not in self._models:
+            raise KeyError(
+                f"No model available for key {key},"
+                f" available models: {self._models.keys()}"
+            )
+
+        X, valid = self._table_to_X(table)
+        n_outputs = getattr(self._models[key], "n_outputs_", 1)
+
+        if n_outputs > 1:
+            shape = (len(table), n_outputs)
+        else:
+            shape = (len(table),)
+
+        prediction = np.full(shape, self.invalid_class, dtype=np.int8)
+        if np.any(valid):
+            prediction[valid] = self._models[key].predict(X)
+
+        return prediction, valid
+
+    def _predict_score(self, key, table):
+        if key not in self._models:
+            raise KeyError(
+                f"No model available for key {key},"
+                f" available models: {self._models.keys()}"
+            )
+
+        X, valid = self._table_to_X(table)
+
+        n_classes = getattr(self._models[key], "n_classes_", 2)
+        n_rows = len(table)
+        shape = (n_rows, n_classes) if n_classes > 2 else (n_rows,)
+
+        scores = np.full(shape, np.nan)
+
+        if np.any(valid):
+            prediction = self._models[key].predict_proba(X)[:]
+
+            if n_classes > 2:
+                scores[valid] = prediction
+            else:
+                # only return one score for the positive class
+                scores[valid] = prediction[:, self._get_positive_index(key)]
+
+        return scores, valid
+
+    def _get_positive_index(self, key):
+        return np.nonzero(self._models[key].classes_ == self.positive_class)[0][0]
+
+
+class EnergyRegressor(SKLearnRegressionReconstructor):
+    """
+    Use a scikit-learn regression model per telescope type to predict primary energy
+    """
+
+    #: Name of the target table column for training
```

Comment:
I think that is self explanatory

Reply:
`#:` makes this a sphinx comment, so it appears in the docs.